app/routes.py

```diff
@@ -1,2 +1,234 @@
+        else: 
+            raise TypeError("Only asc or desc is accepted here")
+    elif filter_by_title:
+        tasks = Task.query.filter_by(title=filter_by_title).all()
+    else:
+        tasks = Task.query.all()
+    tasks_response = [task.to_json() for task in tasks]
+    return jsonify(tasks_response), 200
+  
+@tasks_bp.route("/<task_id>", methods=["GET"], strict_slashes=False)
+def get_single_task(task_id):
+    task = Task.query.get(task_id)
+    if task:
+        return {"task": task.to_json()}, 200
+    else:
+        return jsonify(None), 404
+      
+
+@tasks_bp.route("", methods=["POST"], strict_slashes=False)
+def create_task():
+    request_body = request.get_json()
+    if all(key in request_body for key in ("title", "description", "completed_at")):
+        if validate_data(request_body): 
+            new_task = Task.from_json(request_body)
+            db.session.add(new_task)
+            db.session.commit()
+            return {
+                    "task": new_task.to_json()
+            }, 201
+    else:
+        return {"details": "Invalid data"}, 400
+        
+@tasks_bp.route("/<task_id>", methods=["PUT"], strict_slashes=False)
+def update_task(task_id):
+    task = Task.query.get(task_id)
+    if task:
+        form_data = request.get_json()
+        if validate_data(form_data): 
+            updated_task = Task.from_json(form_data)
+            task.title = form_data["title"]
+            task.description = form_data["description"]
+            task.completed_at = form_data["completed_at"]
+            db.session.commit()
+            return {
+                    "task": task.to_json()
+            }, 200
+    else:
+        return jsonify(None), 404
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"], strict_slashes=False)
+def delete_task(task_id):
+    task = Task.query.get(task_id)
+    if task:
+        db.session.delete(task)
+        db.session.commit()
+        return {
+              "details": f"Task {task.task_id} \"{task.title}\" successfully deleted"
+        }
+    else:
+        return jsonify(None), 404
+      
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"], strict_slashes=False)
+def mark_complete(task_id):
+    task = Task.query.get(task_id)
+    if task:
+        task.completed_at = datetime.utcnow()
+        db.session.commit()
+        send_slack_notifications(task)
+        return {
+        "task": task.to_json()
+        }, 200
+    else:
+        return jsonify(None), 404
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"], strict_slashes=False)
+def mark_incomplete(task_id):
+    task = Task.query.get(task_id)
+    if task:
+        task.completed_at = None
+        db.session.commit()
+        return {
+        "task": task.to_json()
+        }, 200
+    else:
+        return jsonify(None), 404
+
+#################### Routes for Goals ####################
+     
+goals_bp = Blueprint("goals", __name__, url_prefix="/goals") 
```

Comment:
I suggest defining your Blueprints together at the top, just so a user can see which blueprints are available in the routes file.